src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -822,6 +822,9 @@ static void fillQueryEntity(BSONArrayBuilder* baP, const EntityId* enP)
 * If the servicePath is empty, then we return all entities, no matter their servicePath. This
 * can be seen as a query on "/#" considering that entities without servicePath are implicitly
 * assigned to "/" service path.
+*
+* FIXME P2: to gain some tput, if the service path vector is empty (or containing a /#), then the
```

Comment:
In this case, you will get all registrations, instead of the ones using `/#` which is not the aim of:

> Fiware-ServicePath header is taken into account in `GET /v2/subscriptions` and GET /v2/registrations in order to narrow down the results to subscriptions/registrations that use exactly that service path as query.

Reply:
Yeah, more of the same.

New issue to be created for this ...

Reply:
I think we already have an issue for this: https://github.com/telefonicaid/fiware-orion/issues/3051. We closed it time ago, under the understanding that we went through the disussion phase and reach the conclusion that is now driving implementation (and my comments on this PR).

However, if new information have arosen, the issue can be reopened. However, review first all its comments to refresh the memory about this and to be sure the discussion it not already closed and needs to be reopened.

Reply:
Let's just imagine what the user wants when asking for the registrations with Service Path /Madrid/#.
First of all, what does that request mean?
To me, personally, it means that the user wants to see all registrations that have been made with a Service Path that starts with /Madrid, not just those that have the **exact** service path **/Madrid/#**, which would not be very satisfying.

If this "partial functionality" is really what we want, then sure, lets close the entire discussion.
If we instead want to actually respond to the request, and answer with all registrations that fall under /Madrid/#, then a change is needed.

Reply:
I'm a bit confussed... we were talking about moving the discussion to an issue (with that regards I'm proposing an existing one). Thus, why are you adding a new comment (i.e. a new step to the discussion) in the PR intead of using the issue? I'm missing something? :)

Reply:
Easier to just put my opinion right here than in an already closed issue, that's all

Reply:
Issue can be reopened if needed, of course. The only think I ask is to review first the issue history (issue comments) to be sure we are not repeating ourselves.

Reply:
Never mind, I'll just change the PR as requested

Reply:
FIXME removed